### PR TITLE
Improve legibility of the tutorials

### PR DESCRIPTION
**Pull request by hmillerbakewell**
## Notice

- [X] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

## Motivation and Context
To improve legibility of the tutorials

### Explanations of changes
The alterations fit into the following themes
 - typos (e.g. `0,5` -> `0.5`)
 - merging single-sentence paragraphs together into one multi-sentence paragraph
 - using markdown headers rather than forced bold text
 - but using forced bold text to bring attention to a note
 - one very long sentence broken down into a paragraph of sentences

If it was unhelpful to do all these in one go then I trust the CausalNex team to implement those changes they like in the manner they see fit.

## How has this been tested?
Spell checking, no code changes

## Checklist

- [X] Read the [contributing](/CONTRIBUTING.md) guidelines
- [X] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [X] Updated the documentation to reflect the code changes
- [ ] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](/RELEASE.md) file
- [ ] Added tests to cover my changes
- [x] Assigned myself to the PR

### CODE REVIEW

File: docs/source/04_user_guide/04_user_guide.md [hunk 1]
```diff
@@ -216,13 +219,13 @@ by a domain expert. This is a targeted effort to encode important domain knowled
...
 
-> When defining the structure model, we recommend to use the **entire** dataset **without** discretisation of continuous variables.
+> When defining the structure model, we recommend using the **entire** dataset **without** discretisation of continuous variables.
 
 ### 3.2 Likelihood Estimation and Predictions with `BayesianNetwork`
 
 Once the graph has been determined, the `BayesianNetwork` can be initialised and the conditional probability distributions of the variables can be learned from the data.
 
-Maximum likelihood or Bayesian parameter estimation can be used for CPDs learning.
+Maximum likelihood or Bayesian parameter estimation can be used for learning CPDs.
```
Comment (oentaryorj) [inline docs/source/04_user_guide/04_user_guide.md:228]:
```suggestion
Maximum likelihood or Bayesian parameter estimation can be used for learning the CPDs.
```

File: docs/source/04_user_guide/04_user_guide.md [hunk 2]
```diff
@@ -130,30 +130,30 @@ Metrics can help us understand the strength of relationships between variables.
...
 
 The `BayesianNetwork` class is the central class for the causal inference analysis in the package.
-It is built on top of the `StructureModel`, which is an extension of `networkx.DiGraph`
+It is built on top of the `StructureModel` class, which is an extension of `networkx.DiGraph`
 
 `StructureModel` represents a causal graph, a DAG of the respective BN and holds directed edges, describing
 a _cause -> effect_ relationship. In order to define the `BayesianNetwork`, users should provide a relevant `StructureModel`.
 
-> Cycles are permitted within a `StructureModel`. However, only **acyclic connected** `StructureModel` are allowed in the construction of `BayesianNetwork`;  isolated nodes are not allowed.
+> Cycles are permitted within a `StructureModel`. However, only **acyclic connected** `StructureModel` objects are allowed in the construction of `BayesianNetwork`;  isolated nodes are not allowed.
```
Comment (oentaryorj) [inline docs/source/04_user_guide/04_user_guide.md:156]:
```suggestion
> Cycles are permitted within a `StructureModel` object. However, only **acyclic connected** `StructureModel` objects are allowed in the construction of `BayesianNetwork`;  isolated nodes are not allowed.
```

**Review by oentaryorj — APPROVED** [review]
Thank you for vetting our documentation. Approved with a few (minor) suggestions.

**Comment by oentaryorj** [comment]
@hmillerbakewell: Can you also run `make lint` locally to make sure that the linting requirements are met and that our CircleCI pipeline passes? Thank you.

File: docs/source/04_user_guide/04_user_guide.md [hunk 3]
```diff
@@ -186,8 +186,11 @@ For instance, users can define a causal model **fully manually**, e.g., using th
 Or, users can learn the network structure **automatically** from the data using the [`NOTEARS`](https://papers.nips.cc/paper/8157-dags-with-no-tears-continuous-optimization-for-structure-learning.pdf) algorithm. Moreover, if there is domain knowledge available,
 it can be used as **additional constraints** before learning a graph algorithmically.
 
-> Recently published [NOTEARS](https://papers.nips.cc/paper/8157-dags-with-no-tears-continuous-optimization-for-structure-learning.pdf) algorithm for learning DAGs from data based on a continuous optimisation problem
-allowed to overcome the challenges of combinatorial optimisation giving a new impulse to the usage of BNs in the machine learning applications.
+> [NOTEARS](https://papers.nips.cc/paper/8157-dags-with-no-tears-continuous-optimization-for-structure-learning.pdf) 
+> is a recently published algorithm for learning DAGs from data,
+> framed as a continuous optimisation problem.
+> It allowed us to overcome the challenges of combinatorial optimisation,
+> giving a new impetus to the usage of BNs in machine learning applications.
```
Comment (mzjp2) [inline docs/source/04_user_guide/04_user_guide.md:193]:
```suggestion
> [NOTEARS](https://papers.nips.cc/paper/8157-dags-with-no-tears-continuous-optimization-for-structure-learning.pdf) is a recently published algorithm for learning DAGs from data, framed as a continuous optimisation problem. It allowed us to overcome the challenges of combinatorial optimisation, giving a new impetus to the usage of BNs in machine learning applications.
```

don't try and force wrapping for markdown -- let the screen size of the viewer define the wrapping :)

**Review by mzjp2 — APPROVED** [review]
As Richard's comment notes, let's fix the minor suggestions + `make lint` and get this in. Thank you!

**Comment by hmillerbakewell** [comment]
> 
> 
> @hmillerbakewell: Can you also run `make lint` locally to make sure that the linting requirements are met and that our CircleCI pipeline passes? Thank you.

I hadn't appreciated the linting applied to the docs, doing that now.

File: docs/source/04_user_guide/04_user_guide.md (hunk 3 above)
Comment (hmillerbakewell) [reply docs/source/04_user_guide/04_user_guide.md:193]: My mistake

**Comment by hmillerbakewell** [comment]
Running `make install-pre-commit` venv yields `ERROR: Double requirement given: torch<2.0,>=1.4.0 (from -r test_requirements.txt (line 16)) (already in torch<2.0,>=1.7 (from -r requirements.txt (line 9)), name='torch')`. Changing the requirement in test_requirements.txt to `torch >=1.7, <2.0` resolves the error. I'll make an issue for this if I can reliably reproduce it.

**Comment by hmillerbakewell** [comment]
Sadly the installation of pygraphviz fails on this laptop (it doesn't interact well with WSL in my experience), which means I cannot run the necessary make commands required for pre-commit. I'm happy for you to close this pull request and implement the listed changes as your own commits, as that will require less effort overall.

**Comment by mzjp2** [comment]
> Sadly the installation of pygraphviz fails on this laptop (it doesn't interact well with WSL in my experience), which means I cannot run the necessary make commands required for pre-commit. I'm happy for you to close this pull request and implement the listed changes as your own commits, as that will require less effort overall.

Don't worry about it -- the reason the linter was failing was because of a minor whitespace change in the section

> > [NOTEARS](https://papers.nips.cc/paper/8157-dags-with-no-tears-continuous-optimization-for-structure-learning.pdf) is a recently published algorithm for learning DAGs from data, framed as a continuous optimisation problem. It allowed us to overcome the challenges of combinatorial optimisation, giving a new impetus to the usage of BNs in machine learning applications.

which, if you accept my suggestion above, fixes it. So feel free to push the documentation change, and the linters should fix themselves. Otherwise, I'll manually run `make lint` and push to this branch. :)

**Pull request merged**